templates/profiles.html

```diff
@@ -724,33 +722,60 @@ <h4 id="name" style="margin:5px; margin-top:5px;"><span></span></h4>
         return template;
     }
 
-    function buildAbility(ability){
-        let requirements = buildRequirements(ability.test);
-        let template = $("#ability-template").clone();
-        template.attr('id', ability.ability_id)
-            .data('parsers', ability.parsers)
-            .data('testId', ability.ability_id)
-            .data('requirements', requirements);
+    function buildAbility(ability) {
+        let abilityRequirements = [];
+        let cleanupCommands = false;
+        let parsers = false;
+        let payloads = false;
 
+        let template = $('#ability-template').clone();
+        template.attr('id', ability.ability_id)
+            .data('abilityId', ability.ability_id)
+            .data('requirements', abilityRequirements);
         template.find('#name span').html(ability.name);
         template.find('#ability-attack').html(ability.tactic + ' | '+ ability.technique_name);
 
-        if(requirements.length > 0) {
+        ability.executors.forEach((executor) => {
+            if (executor.command !== null) {
+                let executorRequirements = buildRequirements(executor.command);
+                abilityRequirements = abilityRequirements.concat(executorRequirements);
+            }
+
+            cleanupCommands = (cleanupCommands || executor.cleanup.length > 0);
+            parsers = (parsers || executor.parsers.length > 0);
+            payloads = (payloads || executor.payloads.length > 0);
+
+            let executorName = executor.name;
+            if (executorName === 'psh') { executorName = 'powershell'; }
+            else if (executorName === 'pwsh') { executorName = 'powershell core'; }
+            else if (executorName === 'sh') { executorName = 'shell'; }
+            else if (executorName === 'cmd') { executorName = 'commandline'; }
```

Comment:
It looks like this was copied, but this should be done via a map rather than the if/else if chain.